### DISCUSSION TITLE
Unsupervised Learning ?

### DISCUSSION THREAD

**Discussion by lcrmorin** [online-ml/river#450] [Q&A]
I was toying with streaming algorithms in a kaggle competition (https://www.kaggle.com/c/jane-street-market-prediction). The format of this competition is a bit unusual: the training data come as a whole, but the test set is provided in a streaming way. I was wondering if it would be possible to learn from the test set without being provided a target...  

Would the river package allow some sort of unsupervised learning - as in modifying a supervised model - ? Another framework ?

(As I understand the package mostly rely on SGD, so I do not really expect a positive 'clean' answer. But I figured I might ask about more 'dirty' techniques ... some rebalancing past observation to take the new instance into account, learning on the prediction... I don't know)

**Comment by MaxHalford** [comment]
I don't know the ins and outs of that competition. However, I recently did the RIIID! competition, and the premises were the same. I found that updating my feature extractors using the predictions of the model at each batch worked quite well.

River has a lot of cool stuff but it's probably mature enough to be used for what you want to do. Then again, your message is a very open-ended so I'm not sure I'm answering correctly.
